pandas_gbq/gbq.py

```diff
@@ -188,78 +172,103 @@ def get_application_default_credentials(self):
-            # Check if the application has rights to the BigQuery project
-            jobs = bigquery_service.jobs()
-            job_data = {'configuration': {'query': {'query': 'SELECT 1'}}}
-            jobs.insert(projectId=self.project_id, body=job_data).execute()
-            return credentials
-        except:
-            return None
+        Parameters
+        ----------
+        None
 
-    def get_user_account_credentials(self):
-        from oauth2client.client import OAuth2WebServerFlow
-        from oauth2client.file import Storage
-        from oauth2client.tools import run_flow, argparser
+        Returns
+        -------
+        - GoogleCredentials,
+            If the credentials can loaded. The retrieved credentials should
+            also have access to the project (self.project_id) on BigQuery.
+        - OR None,
+            If credentials can not be loaded from a file. Or, the retrieved
+            credentials do not have access to the project (self.project_id)
+            on BigQuery.
+        """
+        with open('bigquery_credentials.dat') as credentials_file:
+            credentials_json = json.load(credentials_file)
+            credentials = google.oauth2.credentials.Credentials(
+                token=credentials_json.get('access_token'),
+                refresh_token=credentials_json.get('refresh_token'),
+                id_token=credentials_json.get('id_token'),
+                token_uri=credentials_json.get('token_uri'),
+                client_id=credentials_json.get('client_id'),
+                client_secret=credentials_json.get('client_secret'),
+                scopes=credentials_json.get('scopes'))
+
+            # Refresh the token before trying to use it.
+            http = httplib2.Http()
+            request = google_auth_httplib2.Request(http)
+            credentials.refresh(request)
+
+        return self.try_credentials(credentials)
+
+    def save_user_account_credentials(self, credentials):
+        """
+        Saves user account credentials to a local file.
+        """
+        with open('bigquery_credentials.dat', 'wb') as credentials_file:
+            credentials_json = {
+                'refresh_token': credentials.refresh_token,
+                'id_token': credentials.id_token,
+                'token_uri': credentials.token_uri,
+                'client_id': credentials.client_id,
+                'client_secret': credentials.client_secret,
+                'scopes': credentials.scopes,
+            }
+            json.dump(credentials_file, credentials_json)
 
-        flow = OAuth2WebServerFlow(
-            client_id=('495642085510-k0tmvj2m941jhre2nbqka17vqpjfddtd'
-                       '.apps.googleusercontent.com'),
-            client_secret='kOc9wMptUtxkcIFbtZCcrEAc',
-            scope=self.scope,
-            redirect_uri='urn:ietf:wg:oauth:2.0:oob')
+    def get_user_account_credentials(self, auth_local_webserver=True):
```

Comment:
are you expecting a user to call this method directly? should this not be in the read_/to_ API?